src/libraries/icubmod/parametricCalibratorEth/parametricCalibratorEth.cpp

```diff
@@ -753,18 +760,32 @@ bool parametricCalibratorEth::calibrate()
         
         if(goneToZero)
         {
-            yDebug() <<  deviceName  << ": set" << setOfJoint_idx  << ": Reached zero position!\n";
+            yDebug() <<  deviceName  << ": set" << setOfJoint_idx  << ": Reached zero position!";
             for(lit  = currentSetList.begin(); lit != currentSetList.end() && !abortCalib; lit++) //for each joint of set
             {
                 iAmp->setPWMLimit((*lit),original_max_pwm[(*lit)]);
             }
         }
-        else          // keep pid safe and go on
+        else // keep pid safe for failed joints and go on
         {
-            yError() <<  deviceName  << ": set" << setOfJoint_idx  << ": some axis got timeout while reaching zero position... disabling this set of axes\n";
-            for(lit  = currentSetList.begin(); lit != currentSetList.end() && !abortCalib; lit++) //for each joint of set
+            yError() <<  deviceName  << ": set" << setOfJoint_idx  << ": some axis got timeout while reaching zero position. Idling failed joints and set on those safe PWM limits";
+            // set the failed joints to idle and set safe PWM limits
+            for (lit = currentSetList.begin(); lit != currentSetList.end() && !abortCalib; lit++)
             {
-                iControlMode->setControlMode((*lit),VOCAB_CM_IDLE);
+                auto it = std::find_if(failedJoints.begin(), failedJoints.end(),
+                                       [lit](int id) { return id == *lit; });
+                
+                if (it != failedJoints.end()) 
+                {
+                    yDebug() << deviceName << ": joint # " << *lit << " failed reaching zero position. Idling it and setting safe PWM limits";
```

Comment:
Hi @MSECode ,
here I suggest to use the yError instead of the yDebug because the failure of reaching the zero position is an error. This could help the user to understand better the problem.